plugins/decklink/decklink-output.cpp

```diff
@@ -32,15 +32,30 @@ static void *decklink_output_create(obs_data_t *settings, obs_output_t *output)
 		DeckLinkDeviceMode *mode =
 			device->FindOutputMode(decklinkOutput->modeID);
 
+		ComPtr<IDeckLinkOutput> output_;
+		device->GetOutput(&output_);
+
+		BMDDisplayMode *actualMode = nullptr;
+		decklink_bool_t modeSupportsBGRA = false;
+		output_->DoesSupportVideoMode(bmdVideoConnectionUnspecified,
+					      mode->GetDisplayMode(),
+					      bmdFormat8BitBGRA,
+					      bmdNoVideoOutputConversion,
+					      bmdSupportedVideoModeDefault,
+					      actualMode, &modeSupportsBGRA);
+
+		const bool supportsHDR = (device->GetSupportsHDRMetadata() &&
+					  !decklinkOutput->force_sdr);
+
 		struct video_scale_info to = {};
-		to.format = VIDEO_FORMAT_BGRA;
+		to.format = (modeSupportsBGRA || supportsHDR)
```

Comment:
Noticed that this code is  "if supports BGR *or* HDR, use BGRA" -- wouldn't this just be "if supports BGR, use BGR"?

Reply:
It's possible I'm misunderstanding this expression though.

Reply:
yeah that probably makes more sense, I think I got confused since HDR was setup for bgra

Reply:
Hmm HDR is using bmdFormat10BitRGBXLE on the device. We should also probably check if that is supported for the current format too. And then figure out like YUV HDR?

Reply:
Probably set this back to always RGB and then do conversions on the GPU here for when we need YUV

https://github.com/obsproject/obs-studio/blob/004253f29633b38e7b1d8e0598e3f91a2ed790a9/UI/frontend-plugins/decklink-output-ui/decklink-ui-main.cpp#L400